File: bakerdelivery/src/main/java/co/simplon/bakerdelivery/service/CommandService.java
package co.simplon.bakerdelivery.service;

import co.simplon.bakerdelivery.dto.CommandDto;
import co.simplon.bakerdelivery.exception.CommandNotFoundException;
import co.simplon.bakerdelivery.model.Command;
import co.simplon.bakerdelivery.model.Etat;
import co.simplon.bakerdelivery.repository.CommandRepository;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;
import java.util.Optional;

@Service
public interface CommandService {

    List<Command> getCommands();

    Optional<Command> getCommandById(Long commandId);

    CommandDto createCommand(CommandDto command);

    CommandDto updateCommand(CommandDto command, Long commandId) throws CommandNotFoundException;

    Boolean deleteCommand(Long commandId);

    List<Command> getCommandsByRestaurant(Long restaurantId);


    Command getCommandByRestaurantIdAndDate( Long restaurantId,LocalDate date) throws CommandNotFoundException;

    List<Command> getCommandsByRestaurantIdAndBetweenTwoDates(Long restaurantId, LocalDate start, LocalDate end);

    List<CommandDto> getCommandsByDate(LocalDate date);

    List<CommandDto> getCommandsByEtatAndDate(Etat etat, LocalDate date);


    List<CommandDto> getCommandsBetweenTwoDates( LocalDate start, LocalDate end);
}

File: lastfront_bakerdelivery/src/app/service/command.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Command} from '../model/command';
import {catchError} from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class CommandService {

  constructor(private httpClient: HttpClient) { }


  getCommandByRestaurantIdAndDate(restaurantId, date): Observable<Command> {
    return this.httpClient.get<Command>
    ('http://localhost:8080/api/bakerdelivery/commands/restaurant/' + restaurantId + '/' + date);
  }

  getCommandsByRestaurantIdAndBetweenTwoDates(restaurantId, start, end): Observable<Command[]> {
    // tslint:disable-next-line:max-line-length
    return this.httpClient.get<Command[]>('http://localhost:8080/api/bakerdelivery/commands/restaurant/' + restaurantId + '/datesbetween?start=' + start + '&end=' + end);
  }


  getCommandsByEtatAndDate(etat, date): Observable<Command[]> {
    return this.httpClient.get<Command[]>('http://localhost:8080/api/bakerdelivery/commands/etat/' + etat + '/' + date);
  }

  getCommandsBetweenTwoDates(id, start, end): Observable<Command[]> {
    return this.httpClient.get<Command[]>('http://localhost:8080/api/bakerdelivery/commands/week/' + id + '/' + start + '/' + end);
  }

  getCommandsByDate(date): Observable<Command[]> {
    return this.httpClient.get<Command[]>('http://localhost:8080/api/bakerdelivery/commands/date/' + date);
  }

  addCommand(command): Observable<Command> {
    return this.httpClient.post<Command>('http://localhost:8080/api/bakerdelivery/commands/', command);
  }

  updateCommand(command, commandId): Observable<Command> {
    return this.httpClient.put<Command>('http://localhost:8080/api/bakerdelivery/commands/' + commandId, command);
  }

  deleteCommand(commandId): Observable<Command> {
    return this.httpClient.delete<Command>('http://localhost:8080/api/bakerdelivery/commands/' + commandId);
  }


}



